app/src/main/java/io/nwhacks/nfc/ReadFragment.java

```diff
@@ -172,6 +170,9 @@ public void onFailure(@NonNull Exception e) {
                             resetDetailView();
                         }
                     });
+        }else {
+            MainActivity.toast(getContext(), "No ApplicantType for this applicant. Please rewrite id.");
```

Comment:
"Please rewrite tag." might be more accurate

Reply:
Besides that, it's just `onEventJoin` that needs to be refactored

Reply:
fixed!